eng/Versions.props

```diff
@@ -2,7 +2,7 @@
 <!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the MIT license. See License.txt in the project root for full license information. -->
 <Project>
   <PropertyGroup>
-    <VersionPrefix>17.13.0</VersionPrefix>
+    <VersionPrefix>17.14.0</VersionPrefix>
     <PackageValidationBaselineVersion>17.12.0</PackageValidationBaselineVersion>
```

Comment:
@rainersigwald do you know if we should also update PackageValidationBaselineVersion?